library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java

```diff
@@ -445,25 +445,37 @@ private long insert(Request request) {
         return ContentUris.parseId(downloadUri);
     }
 
-    public void pauseBatch(long id) {
-        ContentValues values = new ContentValues();
-        String where = COLUMN_BATCH_ID + "= ? AND " + DownloadContract.Downloads.COLUMN_STATUS + " != ?";
-        String[] selectionArgs = {String.valueOf(id), String.valueOf(DownloadStatus.SUCCESS)};
-        values.put(DownloadContract.Downloads.COLUMN_CONTROL, DownloadsControl.CONTROL_PAUSED);
-        contentResolver.update(downloadsUriProvider.getAllDownloadsUri(), values, where, selectionArgs);
-    }
+    public void pauseBatch(long batchId) {
+        DownloadDeleter downloadDeleter = new DownloadDeleter(contentResolver);
+        RealSystemFacade systemFacade = new RealSystemFacade(GlobalState.getContext(), new Clock());
+        BatchRepository batchRepository = new BatchRepository(contentResolver, downloadDeleter, downloadsUriProvider, systemFacade);
 
-    public void resumeBatch(long id) {
-        ContentValues values = getResumeContentValues();
-        String where = COLUMN_BATCH_ID + "= ? AND " + DownloadContract.Downloads.COLUMN_STATUS + " != ?";
-        String[] selectionArgs = {String.valueOf(id), String.valueOf(DownloadStatus.SUCCESS)};
-        contentResolver.update(downloadsUriProvider.getAllDownloadsUri(), values, where, selectionArgs);
+        int batchStatus = batchRepository.getBatchStatus(batchId);
+        if (DownloadStatus.isRunning(batchStatus)) {
```

Comment:
I think it would be nice if this and the pause method were extracted into a class, but not a blocker for me :+1: